parlai/core/torch_ranker_agent.py

```diff
@@ -313,9 +318,29 @@ def eval_step(self, batch):
             cand_preds_generator = (cand_list[rank] for rank in ordering)
             cand_preds.append(list(islice(cand_preds_generator, max_preds)))
 
+        if (self.opt.get('repeat_blocking_heuristic', False) and
+            self.opt.get('eval_candidates') == 'fixed'):
+            cand_preds = self.block_repeats(cand_preds)
+
         preds = [cand_preds[i][0] for i in range(batchsize)]
         return Output(preds, cand_preds)
 
+    def block_repeats(self, cand_preds):
+        """Heuristic to block a model repeating a line from the history."""
+        history_strings = []
```

Comment:
i don't quite understand what this is doing. why do we have to split on new line here?

Reply:
is this for like personas or something?

Reply:
yes for personas and stuff